src/components/Inputs/RadioButtons/RadioButtons.js

```diff
@@ -0,0 +1,293 @@
+  'data-tid': PropTypes.string,
+  onClick: PropTypes.func,
+
+  // These will appear if RadioButtonGroup is used with redux-form:
+  onBlur: PropTypes.func,
+  onChange: PropTypes.func,
+  onDragStart: PropTypes.func,
+  onDrop: PropTypes.func,
+  onFocus: PropTypes.func,
+}
+
+export function RadioButtonGroup({
+  options,
+  onChange,
+  formChangeHandler,
+  allCaps = true,
+  name = `radio-button-group-${uuidv4()}`,
+  initialValue = undefined,
+  currentValue,
+  currentError,
+  formTouched,
+  disabled,
+  validator,
+  required,
+  labelCopy,
+  ...rest
+}) {
+  let initialSelected
+  if (currentValue || typeof currentValue === 'boolean') {
+    initialSelected = currentValue
+  } else if (initialValue || typeof initialValue === 'boolean') {
+    initialSelected = initialValue
+  }
+
+  const [selectedValue, setSelectedValue] = useState(initialSelected)
+  const [isAnswered, setIsAnswered] = useState(false)
+  const resolvedValidator = validator ? validator : () => ''
+  const [getError, setError, , validate] = useErrorMessage(resolvedValidator)
+
+  const validationRadio = () => {
+    let errorMessage = validate(selectedValue)
+    errorMessage = errorMessage.length ? errorMessage : ''
+    setError(errorMessage)
+    if (formChangeHandler) {
+      // Update form with the new value and a falsy error message
+      formChangeHandler(selectedValue, errorMessage)
+    }
+  }
+  useEffect(() => {
+    const isSelectedValue = typeof selectedValue !== 'undefined'
+    if (onChange && isSelectedValue) {
+      onChange({ value: selectedValue, isAnswered })
+      validationRadio()
+    }
+  }, [selectedValue, isAnswered])
+
+  function onClickHandler(value, clickHandler) {
+    return (evt) => {
+      setSelectedValue(value)
+      if (!isAnswered) {
+        setIsAnswered(true)
+      }
+      return clickHandler && clickHandler(evt)
+    }
+  }
+
+  let hasChecked = false
+  const allRadioValues = []
+  const finalOptions = options.map((o) => {
+    const { value, onClick: passedHandler } = o
+    allRadioValues.push(value)
+    const checked = value === selectedValue
+    // Keeps track of whether or not we have any radio checked
+    if (checked) {
+      hasChecked = true
+    }
+    const onClick = onClickHandler(value, passedHandler)
+    return { ...o, name, checked, onClick, onChange }
+  })
+
+  const handleBlur = () => {
+    validationRadio()
+  }
+
+  // This key handling essentially allows us to comply with the spec. See their example:
+  // https://www.w3.org/TR/wai-aria-practices/examples/radio/radio-1/js/radioGroup.js
+  // And Jared Palmer's React version which he obviously referenced w3's example:
+  // https://github.com/palmerhq/radio-group/blob/master/src/index.tsx#L92
+  const handleKeyDown = React.useCallback(
+    (event) => {
+      const last = allRadioValues.length - 1
+      const index = allRadioValues.findIndex((i) => i == selectedValue)
+      event.persist()
+      var flag = false
```

Comment:
nit: could we add a more descriptive name for flag? like 'stateChanged' or something .. not sure this makes sense, that's how I'm reading it.

Reply:
Yeah, `keyActedOn`